File: J2EE_DoANMonHoc_WebTuyenDungViecLam/src/main/java/com/example/DA_WebTuyenDungViecLam/service/JobService.java
package com.example.DA_WebTuyenDungViecLam.service;

import com.example.DA_WebTuyenDungViecLam.dto.request.JobCreateRequest;
import com.example.DA_WebTuyenDungViecLam.dto.request.JobUpdateRequest;
import com.example.DA_WebTuyenDungViecLam.dto.response.JobResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface JobService {

    Page<JobResponse> getPublishedJobs(Pageable pageable, String keyword, Integer categoryId, String city, String jobType);

    JobResponse getById(Long id);

    JobResponse create(JobCreateRequest request, Long employerId);

    JobResponse update(Long id, JobUpdateRequest request, Long employerId);

    void delete(Long id, Long employerId);

    Page<JobResponse> getByEmployer(Long employerId, Pageable pageable);

    JobResponse updateStatus(Long id, String status);

    JobResponse updateStatusByEmployer(Long id, String status, Long employerId);
}

File: J2EE_DoANMonHoc_WebTuyenDungViecLam/frontend/src/services/jobService.ts
import axiosInstance from './axiosInstance';
import type {
  ApiResponse,
  Job,
  JobCreateRequest,
  JobSearchParams,
  JobUpdateRequest,
} from '@/types';

export const jobService = {
  search: (params: JobSearchParams) =>
    axiosInstance.get<ApiResponse<Job[]>>('/jobs', { params }),

  getById: (id: number) =>
    axiosInstance.get<ApiResponse<Job>>(`/jobs/${id}`),

  create: (data: JobCreateRequest) =>
    axiosInstance.post<ApiResponse<Job>>('/jobs', data),

  update: (id: number, data: JobUpdateRequest) =>
    axiosInstance.put<ApiResponse<Job>>(`/jobs/${id}`, data),

  delete: (id: number) =>
    axiosInstance.delete<ApiResponse<void>>(`/jobs/${id}`),

  myJobs: () =>
    axiosInstance.get<ApiResponse<Job[]>>('/jobs/my'),

  updateStatus: (id: number, status: string) =>
    axiosInstance.patch<ApiResponse<Job>>(`/jobs/${id}/status`, { status }),

  updateMyJobStatus: (id: number, status: string) =>
    axiosInstance.patch<ApiResponse<Job>>(`/jobs/my/${id}/status`, { status }),
};
